xmtp_id/src/scw_verifier.rs

```diff
@@ -265,4 +332,112 @@ pub mod tests {
         })
         .await;
     }
+
+    // Testing ERC-6492 with deployed / undeployed coinbase smart wallet(ERC-1271) contracts, and EOA.
+    #[tokio::test]
+    async fn test_erc6492_coinbase_smart_wallet() {
+        with_smart_contracts(|anvil, _provider, client, smart_contracts| async move {
+            // Create owner EOA wallet and then create smart contract wallet account from the factory.
+            let owner: LocalWallet = anvil.keys()[0].clone().into();
+            let owners_addresses = vec![Bytes::from(H256::from(owner.address()).0.to_vec())];
+            let factory = smart_contracts.coinbase_smart_wallet_factory();
+            let nonce = U256::from(0);
+            let smart_wallet_address = factory
+                .get_address(owners_addresses.clone(), nonce)
+                .await
+                .unwrap();
+            let contract_call = factory.create_account(owners_addresses.clone(), nonce);
+            let pending_tx = contract_call.send().await.unwrap();
+            pending_tx.await.unwrap();
+
+            assert!(
+                is_smart_contract(smart_wallet_address, anvil.endpoint(), None)
+                    .await
+                    .unwrap()
+            );
+
+            // Generate the signature for coinbase smart wallet
+            let smart_wallet = CoinbaseSmartWallet::new(
+                smart_wallet_address,
+                Arc::new(client.with_signer(owner.clone().with_chain_id(anvil.chain_id()))),
+            );
+            let hash: [u8; 32] = H256::random().into();
+            let replay_safe_hash = smart_wallet.replay_safe_hash(hash).call().await.unwrap();
+            let signature = owner.sign_hash(replay_safe_hash.into()).unwrap();
+            let signature: Bytes = abi::encode(&[Token::Tuple(vec![
+                Token::Uint(U256::from(0)),
+                Token::Bytes(signature.to_vec()),
+            ])])
+            .into();
+
+            let verifier = SmartContractWalletVerifier::new(anvil.endpoint());
+
+            // As a precheck, it ensures that the signature is valid on ERC-1271
+            assert!(verifier
+                .erc1271_is_valid_signature(smart_wallet_address, hash, signature.clone(), None,)
+                .await
+                .unwrap());
+
+            // Testing ERC-6492 signatures with deployed ERC-1271.
+            assert!(verifier
+                .erc6492_is_valid_signature(smart_wallet_address, hash, &signature, None,)
+                .await
+                .unwrap());
+
+            assert!(!verifier
+                .erc6492_is_valid_signature(
+                    smart_wallet_address,
+                    H256::random().into(),
+                    &signature,
+                    None,
+                )
+                .await
+                .unwrap());
+
+            // TODO: Testing ERC-6492 signatures with undeployed ERC-1271.
```

Comment:
It's a little bit tricky to generate signatures for undeployed ERC-1271. Though I consider it's fine to left a TODO because Viem doesn't test this either and it's recommended to be used by Coinbase.